File: CSCI-3800-Advanced-Programming/HW04/TateP_Problem4/PrintedBook.py
# Patrick Tate
# CSCI 3800 HW 4 Problem 4

from Book import Book # because PrintedBook inherits from base class Book

# simple class to model a sublcass of Book called PrintedBook
# PrintedBook has all member variables from Book
# PrintedBook also has a member variable named "type" to denote either a "hard cover" or "paperback"
class PrintedBook(Book):

    def __init__(self, title="", location="", yearPub=0, authors=[], publisher="", pages=0, type=""):
        super(PrintedBook, self).__init__(title, location, yearPub, authors, publisher, pages)
        self.__type = type

    def getType(self):
        return self.__type

    def setType(self, type):
        self.__type = type

    # overrides the abstract method printBook from the abstract base class Book
    def printBook(self):
        print("Title: " + str(self.getTitle()) + ". " + "Author/s: " + str(self.getAuthorsString()))
        print("Publisher: " + self.getPublisher().getName() + ". Publication year: " + str(self.getYearPub()) + ".")
        print("Pages: " + str(self.getPages()) + "." + " Type: " + self.__type)

File: CSCI-3800-Advanced-Programming/HW04/TateP_Problem2/src/PrintedBook.java
/*
    - Patrick Tate
    - CSCI 3800 HW 4 Problem 2

 */

import java.util.ArrayList;
import java.util.HashSet;
import java.util.*;

public class PrintedBook extends Book {

    private int totalPages;
    private String type; // paperback or hardcover

    // default constructor
    public PrintedBook(){
        this.totalPages = 0;
        this.type = "";
    }

    public PrintedBook(String title, String location, int yearPub,
                       HashSet<Author> authorList, Publisher publisher,
                       int pages, String type){
        super(title, location, yearPub, authorList, publisher);

        this.totalPages = pages;
        this.type = type;

    }

    // getters
    public int getTotalPages(){
        return this.totalPages;
    }
    public String getType(){
        return this.type;
    }

    // setters
    public void setTotalPages(int pages){
        this.totalPages = pages;
    }
    public void setType(String type){
        this.type = type;
    }

    // overrides toString from base Book class
    @Override
    public String toString(){

        return "Title: '" + this.getTitle() + ".'" + " Authors: " + this.getAuthors() + "\n" +
                "Total Pages: " + this.totalPages + "." + " Type: " + this.type + "\n" +
                "Publisher: " + this.getPublisher() + "." + " Publication Year: " + this.getYearPub() + "\n";
    }

}
